modules/pubsub/broker.go

```diff
@@ -0,0 +1,101 @@
+package pubsub
+
+import (
+	"sync"
+
+	"github.com/spaceuptech/space-cloud/model"
+	"github.com/spaceuptech/space-cloud/utils"
+)
+
+natsWildcard string = ">"
+
+type subscription interface {
+	Unsubscribe() error
+}
+
+type pubsubBroker interface {
+	Publish(subject string, msg *model.PubsubMsg) error
+	QueueSubscribe(subject, group string, ch chan *model.PubsubMsg) (*subscription, error)
+}
+
+type pubsubSubscription {
+	subs *subscription
+	ch   chan *model.PubsubMsg
+}
+
+func (m *Module) Publish(subject string, data *model.PubsubMsg) error {
+	return m.connection.Publish(subject, data)
+}
+
+func (m *Module) Subscribe(clientId, subject string, cb model.PubsubCallback) (error) {
+	return m.QueueSubscribe(clientId, subject, clientId, cb)
```

Comment:
The arguments seem to be incorrect. You are passing 4 paremeters whereas it requires only 3. Have you build the code? And shouldn't the subject here be also single leading?

Reply:
This is correct @Jayesh333

Reply:
Yess